orangecontrib/prototypes/widgets/owattributes.py

```diff
@@ -0,0 +1,394 @@
+        x_max, x_min = x.max(), x.min()
+        # Prepare the binary encoded y matrix
+        n_classes = len(data.domain.class_var.values)
+        n_samples = x.shape[0]
+        y = np.zeros((n_samples, n_classes))
+        y[np.arange(n_samples), data.Y.astype(np.int64)] = 1
+
+        # Add 2 to account for ignoring both extreme edges
+        edges = np.linspace(x_min, x_max, self.n_bins)[1:]
+        indices = np.digitize(x, bins=edges)
+        bin_samples = np.bincount(indices)
+        # TODO This will fail on a continuous class variable
+        # If the class is discrete, we can compute the class distribution
+        # inside every bin
+        distributions = np.zeros((self.n_bins, n_classes))
+        for bin_idx in range(self.n_bins):
+            mask = indices == bin_idx
+            distributions[bin_idx] = y[mask].sum(axis=0)
+
+        # Add the starting and ending edge
+        edges = np.linspace(x_min, x_max, self.n_bins)
+        edge_diff = edges[1] - edges[0]
+        edges = np.hstack((edges, [edges[-1] + edge_diff]))
+
+        return edges, bin_samples, distributions
+
+    def _draw_histogram(self):
+        _, bin_samples, distributions = self._calculate_histogram(self.data)
+
+        for n_samples, distribution in zip(bin_samples, distributions):
+            bar = HistogramBarItem(n_samples, distribution, self.data.domain,
+                                   histogram=self)
+            self.__layout.addItem(bar)
+
+    def boundingRect(self):
+        return QRectF(0, 0, self.width, self.height)
+
+    def paint(self, painter, option, widget):
+        super().paint(painter, option, widget)
+
+
+class AttributeRow:
+    # Name is used for header, description for tooltip, field can be one of:
+    # - attribute on `AttributeRow` or raw attribute
+    # - method on `AttributeRow` that takes no parameters
+    # - any identifier that server no extra function e.g. icon
+    Field = namedtuple('Field', ['name', 'description', 'field'])
+
+    FIELDS = [
+        Field('', '', 'icon'),
+        Field('Name', '', 'name'),
+        Field('Distribution', '', 'distribution'),
+        Field('First moment', '', 'first_moment'),
+        Field('Second moment', '', 'second_moment'),
+        Field('Valid', '', 'valid'),
+    ]
+
+    def __init__(self, attribute, data):
+        self.attribute = attribute
+        self._data = data
+
+    def __getitem__(self, item):
+        assert isinstance(self.FIELDS[item], self.Field)
+        if self.FIELDS[item].field and hasattr(self, self.FIELDS[item].field):
+            result = getattr(self, self.FIELDS[item].field)
+            return result() if callable(result) else result
+
+    def __getattr__(self, item):
+        """Fallback to the attribute object field if it exists."""
+        if hasattr(self.attribute, item):
+            return getattr(self.attribute, item)
+        else:
+            raise AttributeError('Could not find `%s`' % item)
```

Comment:
Yes, that's true, but then in case the error does actually get raised, the AttributeError message will refer to the `self.attribute`, which can be very confusing. This way, it is clear where the error occurs.